### PR TITLE
Migrate some tests away from kIgnoreCertificateErrors

### PR DESCRIPTION
**Pull request by pilgrim-brave**
Resolves https://github.com/brave/brave-browser/issues/19418
Resolves https://github.com/brave/brave-browser/issues/19417
Resolves https://github.com/brave/brave-browser/issues/19415
Resolves https://github.com/brave/brave-browser/issues/19414
Resolves https://github.com/brave/brave-browser/issues/19413
Resolves https://github.com/brave/brave-browser/issues/19410
Resolves https://github.com/brave/brave-browser/issues/19419
Resolves https://github.com/brave/brave-browser/issues/19408
Resolves https://github.com/brave/brave-browser/issues/19409

Resolves https://github.com/brave/brave-browser/issues/19412
Note: some changes to Shields API tests were necessary because they share a test file (`test/data/load_js_from_origins.html`)

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by bbondy** [comment]
Rebased to latest master because it was failing on applying patches.

**Comment by goodov** [comment]
[This is what I get](https://gist.github.com/goodov/a0e620a7859bcee7bd444f648e71fb92) when I search for `kIgnoreCertificateErrors`. Looks like this PR doesn't include all test files. Can you please create an issue so we can deal with left files at some point and close it when we're done?

File: browser/permissions/permission_lifetime_manager_browsertest.cc [hunk 1]
```diff
@@ -55,17 +56,27 @@ const char kPreTestDataFileName[] = "pre_test_data";
...
+    InProcessBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::TearDownInProcessBrowserTestFixture();
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
   }
 
   void SetUpOnMainThread() override {
```
Comment (goodov) [inline browser/permissions/permission_lifetime_manager_browsertest.cc:78]: let's call parent class here to be safe: `InProcessBrowserTest::SetUpOnMainThread();`

File: browser/permissions/permission_manager_browsertest.cc [hunk 2]
```diff
@@ -76,13 +76,9 @@ class PermissionManagerBrowserTest : public InProcessBrowserTest {
 
   ~PermissionManagerBrowserTest() override = default;
 
-  void SetUpCommandLine(base::CommandLine* command_line) override {
-    InProcessBrowserTest::SetUpCommandLine(command_line);
-    command_line->AppendSwitch(switches::kIgnoreCertificateErrors);
-  }
-
   void SetUpOnMainThread() override {
```
Comment (goodov) [inline browser/permissions/permission_manager_browsertest.cc:79]: and here `InProcessBrowserTest::SetUpOnMainThread();`

**Review by goodov — APPROVED** [review]
++ but please add parent class calls in few places

File: browser/permissions/permission_manager_browsertest.cc (hunk 2 above)
Comment (pilgrim-brave) [reply browser/permissions/permission_manager_browsertest.cc:79]: added in https://github.com/brave/brave-core/commit/52c05a3594dbb439663b7b100cc719d80f3cb7db

File: browser/permissions/permission_lifetime_manager_browsertest.cc (hunk 1 above)
Comment (pilgrim-brave) [reply browser/permissions/permission_lifetime_manager_browsertest.cc:78]: added in https://github.com/brave/brave-core/commit/52c05a3594dbb439663b7b100cc719d80f3cb7db

**Comment by pilgrim-brave** [comment]
> [This is what I get](https://gist.github.com/goodov/a0e620a7859bcee7bd444f648e71fb92) when I search for `kIgnoreCertificateErrors`. Looks like this PR doesn't include all test files. Can you please create an issue so we can deal with left files at some point and close it when we're done?

Opened https://github.com/brave/brave-browser/issues/19456

**Review by iefremov — APPROVED** [review]

**Pull request merged**